components/ui/pathogen-map/esm-maps.tsx

```diff
@@ -0,0 +1,46 @@
+import { ArboContext } from "@/contexts/arbo-context/arbo-context";
+import { useContext } from "react";
+import { Layer, Source } from "react-map-gl";
+
+type Esms = "zika" | "dengue2015" | "dengue2050";
+
+const esmIds: Record<Esms, string> = {
+  zika: "zika_esm_source",
+  dengue2015: "dengue_2015_esm_source",
+  dengue2050: "dengue_2050_esm_source",
+};
+
+const esmUrls: Record<Esms, string> = {
+  zika: "mapbox://serotracker.7qe2zkgd",
+  dengue2015: "mapbox://serotracker.5wobnon2",
+  dengue2050: "mapbox://serotracker.9lzerqbc",
+};
+
+export function EsmMapSourceAndLayer(props: {popupLayerId?: string}) {
+  const state = useContext(ArboContext);
+
+  if(!state.selectedFilters.esm || state.selectedFilters.esm.length === 0) return null;
+
+  const selectedEsms = state.selectedFilters.esm[0];
+
+  return (
+    <Source
+      id={"esm_source"}
+      type="raster"
+      url={esmUrls[selectedEsms as Esms]}
+    >
+      <Layer
+        id={"zika_esm_source"}
```

Comment:
Did you mean to use `esmIds` here? I think `esmIds` is usused otherwise, right? Would be down to just give this some arbitrary id and remove `esmIds`.